lineage.cue

```diff
@@ -165,7 +165,7 @@ import (
 	// _basis keeps the index of the first schema in each major version
 	// within the overall canonical schema sort ordering. This allows trivial
 	// schema retrieval from a syntactic version.
-	_basis: [0, for maj, _ in list.Drop(_counts, len(_counts)-1) {
+	_basis: [0, for maj, _ in list.Drop(_counts, 1) {
 		list.Sum(list.Take(_counts, maj+1))
 	}]
```

Comment:
```suggestion
	_basis: [0, for i, sch in list.Drop(SS, 1) if SS[i].version[0] < sch.version[0] {i + 1}]
```

Looking at the code around, looks to me that the `_pos` variable defined [here](https://github.com/grafana/thema/blob/main/lineage.cue#L148) (_as in the suggestion_) does exactly the same, in a way that imho it's even easier to read & understand.

So, if my assumption is correct, I'd suggest to define it that way, and we might also be able to reuse it and get rid of one variable / calc. Wdyt? 🤔